src/api/firebase.js

```diff
@@ -221,3 +226,107 @@ export async function deleteItem(listPath, itemId) {
 		throw new Error(`Failed updating item: ${error.message}`);
 	}
 }
+
+export let urgencyObject = {
+	overdue: new Set(),
+	soon: new Set(),
+	kindOfSoon: new Set(),
+	notSoon: new Set(),
+	inactive: new Set(),
+};
+
+/**
+ * Sorts an item into one of four urgency categories based on the provided urgency status.
+ *
+ * Items are categorized into 'soon', 'kindOfSoon', 'notSoon', or 'inactive' based on the urgency.
+ *
+ * @param {Object} item - The item to be categorized. Should include at least a `name` property.
+ * @param {number} urgencyStatus - The urgency status of the item, used to determine its category.
+ * @throws Will throw an error if the item cannot be placed in a category.
+ */
+const sortByUrgency = (item, daysUntilNextPurchase) => {
+	if (item.name.includes('sun')) {
+		console.log(`${item.name} urgencyStatus ${daysUntilNextPurchase}`);
+	}
+	if (daysUntilNextPurchase < 0) {
+		urgencyObject.overdue.add(item);
+		return;
+	} else if (daysUntilNextPurchase === 1000) {
+		urgencyObject.inactive.add(item);
+		return;
+	} else if (daysUntilNextPurchase < 7) {
+		urgencyObject.soon.add(item);
+		return;
+	} else if (daysUntilNextPurchase >= 7 && daysUntilNextPurchase < 30) {
+		urgencyObject.kindOfSoon.add(item);
+		return;
+	} else if (daysUntilNextPurchase >= 30) {
+		urgencyObject.notSoon.add(item);
+		return;
+	} else {
+		throw new Error(`Failed to place [${item.name}]`);
+	}
+};
```

Comment:
This function appears to be purely client-side and doesn't interact with Firebase. Could we move it to a utility file and add a test to ensure its logic is covered?